src/Umbraco.Web/PropertyEditors/GridPropertyEditor.cs

```diff
@@ -22,25 +24,74 @@ namespace Umbraco.Web.PropertyEditors
+        }
 
         public override IPropertyIndexValueFactory PropertyIndexValueFactory => new GridPropertyIndexValueFactory();
 
         /// <summary>
         /// Overridden to ensure that the value is validated
         /// </summary>
         /// <returns></returns>
-        protected override IDataValueEditor CreateValueEditor() => new GridPropertyValueEditor(Attribute);
+        protected override IDataValueEditor CreateValueEditor() => new GridPropertyValueEditor(Attribute, _mediaService, _contentTypeBaseServiceProvider);
 
         protected override IConfigurationEditor CreateConfigurationEditor() => new GridConfigurationEditor();
 
         internal class GridPropertyValueEditor : DataValueEditor
         {
-            public GridPropertyValueEditor(DataEditorAttribute attribute)
+            private IMediaService _mediaService;
+            private IContentTypeBaseServiceProvider _contentTypeBaseServiceProvider;
+
+            public GridPropertyValueEditor(DataEditorAttribute attribute, IMediaService mediaService, IContentTypeBaseServiceProvider contentTypeBaseServiceProvider)
                 : base(attribute)
-            { }
+            {
+                _mediaService = mediaService;
+                _contentTypeBaseServiceProvider = contentTypeBaseServiceProvider;
+            }
+
+            /// <summary>
+            /// Format the data for persistence
+            /// This to ensure if a RTE is used in a Grid cell/control that we parse it for tmp stored images
+            /// to persist to the media library when we go to persist this to the DB
+            /// </summary>
+            /// <param name="editorValue"></param>
+            /// <param name="currentValue"></param>
+            /// <returns></returns>
+            public override object FromEditor(ContentPropertyData editorValue, object currentValue)
+            {
+                if (editorValue.Value == null)
+                    return null;
+
+                // editorValue.Value is a JSON string of the grid
+                var rawJson = editorValue.Value.ToString();
+                var grid = JsonConvert.DeserializeObject<GridValue>(rawJson);
+
+                // Find all controls that use the RTE editor
+                var controls = grid.Sections.SelectMany(x => x.Rows.SelectMany(r => r.Areas).SelectMany(a => a.Controls));
+                var rtes = controls.Where(x => x.Editor.Alias.ToLowerInvariant() == "rte");
+
+                foreach(var rte in rtes)
+                {
+                    // Parse the HTML
+                    var html = rte.Value?.ToString();
+                    
+                    var userId = Current.UmbracoContext.Security.CurrentUser.Id;
+
+                    // TODO: In future task(get the parent folder from this config) to save the media into
+                    var parsedHtml = TemplateUtilities.FindAndPersistPastedTempImages(html, -1, userId, _mediaService, _contentTypeBaseServiceProvider);
```

Comment:
```suggestion
                    var parsedHtml = TemplateUtilities.FindAndPersistPastedTempImages(html, Constants.System.Root, userId, _mediaService, _contentTypeBaseServiceProvider);
```

Don't introduce more magic integers